### PR TITLE
fix(CORE): another attempt to fix the dbimport command line warning

### PR DESCRIPTION
**Pull request by Yehonal**
## Changes Proposed:
-  This new approach uses the defaults-extra-file strategy mysql argument to provide a password through a config file that is created by the server on the fly
- Implemented the TempDir config that can be used to store temporary files (including the generated mysql conf)

Related: https://github.com/azerothcore/azerothcore-wotlk/pull/13404

## Tests Performed:

- Running on a linux and windows environment. No errors

## How to Test the Changes:

1. Run the worldserver preferably on a new database 
2. check if you have any error

### CODE REVIEW

**Comment by Shauren** [comment]
Wouldn't it be more user-friendly if temporary path wouldn't have to be set in config?

I was thinking about simply using `boost::filesystem::path tempFile = boost::filesystem::temp_directory_path() / boost::filesystem::unique_path()` to create the file (then delete it after updater runs)

**Comment by Yehonal** [comment]
> Wouldn't it be more user-friendly if temporary path wouldn't have to be set in config?
> 
> I was thinking about simply using `boost::filesystem::path tempFile = boost::filesystem::temp_directory_path() / boost::filesystem::unique_path()` to create the file (with linux permissions 600, delete it after updater runs)

If you run AzerothCore under docker or any other read-only filesystem, it's possible that the temp path is not writeable while you have a different read-write mounted folder/volume. 
But indeed, we should probably use the temp_directory_path as default when the conf is not set by the user

**Comment by Yehonal** [comment]
> Isn't that simply user responsibility? `temp_directory_path` doesn't simply return a hardcoded path but queries a bunch of environment variables to obtain temporary path
> 
> Both std::filesystem and boost::filesystem behave in this way linux: `he path supplied by the first environment variable found in the list TMPDIR, TMP, TEMP, TEMPDIR. If none of these are found, "/tmp"` windows: `The path reported by the Windows GetTempPath API function`

if you have access to the OS that's possible, but in a more managed environment, you can't always do that.

However, this approach should allow all the cases mentioned:
https://github.com/azerothcore/azerothcore-wotlk/pull/13449/commits/ab8e3c18924f73092cf0b2bcbe75f4fa932435f6

**Comment by Yehonal** [comment]
> Maybe I'm missing something here but if you can run the container yourself (which you have to be able to do to mount volumes with extracted data) then you can also specify --tmpfs (https://docs.docker.com/storage/tmpfs/) and obtain usable /tmp inside the container

I'm just considering a possible environment where you can neither mount a folder nor change the temp directory and are forced to use a specific path. Or, maybe the user wants to put those AC temporary files somewhere else rather than in the root of the /tmp folder, to use a different mount type, organise them better, let them be accessible etc.

The YAGNI pattern is not applicable here while implementing this possibility can be helpful for specific users. 

In my case, for instance, I want to put them in a repository folder but I do not want to change the entire OS TMP env variable to do that.

**Comment by Winfidonarleyan** [comment]
1. `MariaDB` does not give out this warning, so these changes are only necessary for `MySQL`.
2. For `Ubuntu/Debian` there is already a ready-made config file `/etc/mysql/debian.cnf`.

----

I made a like implementation in my core
Example dry run - https://github.com/WarheadCore/WarheadBand/actions/runs/3278454336/jobs/5396856110

File: src/server/database/Updater/DBUpdater.cpp [hunk 1]
```diff
@@ -440,16 +440,29 @@ template<class T>
...
+        tempDir = std::filesystem::temp_directory_path();
+
+    tempDir = Acore::String::AddSuffixIfNotExists(tempDir, '/');
+    std::string confFileName = "mysql_ac.conf";
+
+    std::ofstream outfile (tempDir + confFileName);
+
+    outfile << "[client]\npassword = \"" << password << '"' << std::endl;
+
+    outfile.close();
```
[the remark this answers is not in the document]
Comment (Yehonal) [reply src/server/database/Updater/DBUpdater.cpp:455]: If the user changes the password this file should be regenerated. Better to do it always

**Comment by Yehonal** [comment]
> 1. `MariaDB` does not give out this warning, so these changes are only necessary for `MySQL`.
> 2. For `Ubuntu/Debian` there is already a ready-made config file `/etc/mysql/debian.cnf`.
> 
> I made a like implementation in my core - https://github.com/WarheadCore/WarheadBand/blob/0d0f5e4c8df58c37c5fad0c7788c91e4a77abce9/src/server/database/Updater/DBUpdater.cpp#L359 Example dry run - https://github.com/WarheadCore/WarheadBand/actions/runs/3278454336/jobs/5396856110
> 
> Regarding changes, I would make a ready-made config that you just copy to the right place

If you set the password under the etc folder that will be used in all the OS. but we want to use it only for the mysql running for the AC

**Pull request merged**